addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -292,7 +292,11 @@ func (wh *ClusterBootstrap) getGVR(gk schema.GroupKind) (*schema.GroupVersionRes
 	}
 	apiResourceList, err := wh.cachedDiscoveryClient.ServerPreferredResources()
 	if err != nil {
-		return nil, err
+		// Sometimes it is unable to retrieve the complete list of server APIs i.g. an aggregated API server is down,
+		// ErrGroupDiscoveryFailed is returned in this case. We can still get other API resources and ignore this error.
+		if _, ok := err.(*discovery.ErrGroupDiscoveryFailed); !ok {
```

Comment:
Also is this check precise enough to only catch the Metrics server goes down because kubelet certs. 
Would other failures possibly sneak by due to this?  need a unit test for that as well.

Reply:
apiResourceList will be at least an empty list according to implementation in method ServerPreferredResources(). I agree to check nil in case there are changes in the future.